src/Moq/ExpressionExtensions.cs

```diff
@@ -284,8 +284,8 @@ void Split(Expression e, out Expression r /* remainder */, out InvocationShape p
 						r = memberAccessExpression.Expression;
 						var parameter = Expression.Parameter(r.Type, r is ParameterExpression ope ? ope.Name : ParameterName);
 						var property = memberAccessExpression.GetReboundProperty();
-						var method = property.CanRead ? property.GetGetMethod(true) : property.GetSetMethod(true);
-						//                    ^^^^^^^                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+						var method = property.CanRead(out var getter) ? getter : property.GetSetMethod(true);
```

Comment:
Using `property.GetSetMethod(true)` directly should be safe here, since there must be a setter if the property isn't readable. (Properties must have at least one accessor.)